modular_skyrat/modules/holomap/code/station_holomap.dm

```diff
@@ -0,0 +1,304 @@
+		set_light(1, 2)
+
+/obj/machinery/station_map/proc/set_broken()
+	machine_stat |= BROKEN
+	update_icon()
+
+/obj/machinery/station_map/update_icon()
+	. = ..()
+	if(!holomap_datum)
+		return //Not yet.
+
+	cut_overlays()
+	if(machine_stat & BROKEN)
+		icon_state = "station_map_broken"
+	else if(panel_open)
+		icon_state = "station_map_opened"
+	else if((machine_stat & NOPOWER) || !anchored)
+		icon_state = "station_map_off"
+	else
+		icon_state = initial(icon_state)
+
+		if(bogus)
+			holomap_datum.initialize_holomap_bogus()
+		else
+			small_station_map = image(SSholomaps.extra_holomaps["[HOLOMAP_EXTRA_STATIONMAPSMALL]_[current_z_level]"], dir = src.dir)
+			add_overlay(small_station_map)
+
+	// Put the little "map" overlay down where it looks nice
+	if(floor_markings)
+		floor_markings.dir = src.dir
+		floor_markings.pixel_x = -src.pixel_x
+		floor_markings.pixel_y = -src.pixel_y
+		add_overlay(floor_markings)
+
+/obj/machinery/station_map/screwdriver_act(mob/living/user, obj/item/tool)
+	. = default_deconstruction_screwdriver(user, "station_map_opened", "station_map_off", tool)
+	close_map()
+	update_icon()
+
+	if(!.) // Prevent re-running setup holomap for no reason.
+		return FALSE
+
+	if(!panel_open)
+		setup_holomap()
+
+	return .
+
+/obj/machinery/station_map/multitool_act(mob/living/user, obj/item/tool)
+	if(!panel_open)
+		to_chat(user, span_warning("You need to open the panel to change the [src]'[p_s()] settings!"))
+		return TRUE
+	if(!SSholomaps.valid_map_indexes.len > 1)
+		to_chat(user, span_warning("There are no other maps available for [src]!"))
+		return TRUE
+
+	tool.play_tool_sound(user, 50)
+	var/current_index = SSholomaps.valid_map_indexes.Find(current_z_level)
+	if(current_index >= SSholomaps.valid_map_indexes.len)
+		current_z_level = SSholomaps.valid_map_indexes[1]
+	else
+		current_z_level = SSholomaps.valid_map_indexes[current_index + 1]
+
+	to_chat(user, span_info("You set the [src]'[p_s()] database index to [current_z_level]."))
+	return TRUE
+
+/obj/machinery/station_map/crowbar_act(mob/living/user, obj/item/tool)
+	return default_deconstruction_crowbar(tool)
+
+/obj/machinery/station_map/wrench_act(mob/living/user, obj/item/tool)
+	if(!panel_open)
+		return FALSE
+	rotate_map(-90)
+	tool.play_tool_sound(user, 50)
+	return TRUE
+
+/obj/machinery/station_map/wrench_act_secondary(mob/living/user, obj/item/tool)
+	if(!panel_open)
+		return FALSE
+	rotate_map(90)
+	tool.play_tool_sound(user, 50)
+	return TRUE
+
+/// Rotates the map machine by the given amount of degrees. See byond's builtin `turn` for more info.
+/obj/machinery/station_map/proc/rotate_map(direction)
+	dir = turn(dir, direction)
+	switch(dir)
+		if(NORTH)
+			pixel_x = 0
+			pixel_y = 32
+		if(SOUTH)
+			pixel_x = 0
+			pixel_y = -32
+		if(EAST)
+			pixel_x = 32
+			pixel_y = 0
+		if(WEST)
+			pixel_x = -32
+			pixel_y = 0
+
+	setup_holomap() // Required to refresh the small map icon.
+
+/obj/machinery/station_map/emp_act(severity)
+	set_broken()
```

Comment:
guaranteed? will fixing it be painful?

Reply:
Literally just deconstruct and rebuild for now. I'll likely add a flag to the circuit on drop that can be fixed with a multitool or something.